WordPress/wordpress-playground
# Add sites from the site manager

## Motivation for the change, related issues

The current site manager allows users to switch between sites only, but it still isn't clear how to add a site.

To allow users to add sites this PR adds an _Add site_ button and modal for inserting the site name.
On submit a new site is added and loaded.

## Implementation details

The `add-site-button` component validates the name and generates a slug out of it.
Because we can only store the slug until site storage is completed, the name is limited to lowercase letters and numbers.

The site sidebar now supports adding sites. Sites are added to the state which will later get moved to site storage.

The `add-site-button` component isn't completed, this is only a temporary implementation.
Once we have the site manager, the button will just create a new site without any modals or inputs.

## Testing Instructions (or ideally a Blueprint)

- open [Playground with browser storage](http://127.0.0.1:5400/website-server/?storage=browser)
- open the site manager menu (upper left corner)
- see the _Add site_ button
- click on the button and see the modal
- add a site using the modal


![Screenshot 2024-08-09 at 06 57 28](https://github.com/user-attachments/assets/207c416f-af33-4674-b998-a2918dce49db)
